Description of AnnData contents

allyhawkins opened the pull request:
Closes #145 

This PR expands the current section on gene expression file contents to include both SCE and AnnData objects. 

- I created a short introduction section that states gene expression data is available as either SCE or AnnData. Then I created two large sections, one for SCE and one for AnnData (as mentioned in the outline of #145). 
- The two sections mostly mirror each other and each have a description of cell metrics, gene expression metrics, metadata, dim reductions, etc. Although the actual sections mirror each other, I tried to avoid repeating things that are present in both objects. For the cell and row metadata in AnnData objects, I provided links to the SCE section for things that are found in both and then a separate table for any additional information that is present in only AnnData objects. 
- Generally, a lot of the content and descriptions are similar between the AnnData and SCE sections. I was trying to make sure that things were consistent between the two sections.

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -1,21 +1,23 @@
 
-Single-cell or single-nuclei gene expression data (unfiltered, filtered, or processed) is provided for use with R as an RDS file containing a [`SingleCellExperiment` object](http://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html).
-This object contains the expression data, cell and gene metrics, associated metadata, and, in the case of multimodal data like ADTs from CITE-seq experiments, data from additional cell-based assays.
+Single-cell or single-nuclei gene expression data (unfiltered, filtered, or processed) is provided for use with R as an RDS file containing a [`SingleCellExperiment` object](http://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html) or with python as an HDF5 file containing an [`AnnData` object](https://anndata.readthedocs.io/en/latest/index.html).
```
Would this be a little easier to consume with nested bullets? For example (apologies for almost-certainly-bad suggestion spacing...)

```suggestion
Single-cell or single-nuclei gene expression data (unfiltered, filtered, or processed) is provided in two formats:
  - As an RDS file containing a [`SingleCellExperiment` object](http://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html) for use in R
 - An HDF5 file containing an [`AnnData` object](https://anndata.readthedocs.io/en/latest/index.html) for use in Python
```

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -1,21 +1,23 @@
-Single-cell or single-nuclei gene expression data (unfiltered, filtered, or processed) is provided for use with R as an RDS file containing a [`SingleCellExperiment` object](http://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html).
-This object contains the expression data, cell and gene metrics, associated metadata, and, in the case of multimodal data like ADTs from CITE-seq experiments, data from additional cell-based assays.
+Single-cell or single-nuclei gene expression data (unfiltered, filtered, or processed) is provided for use with R as an RDS file containing a [`SingleCellExperiment` object](http://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html) or with python as an HDF5 file containing an [`AnnData` object](https://anndata.readthedocs.io/en/latest/index.html).
+These objects contain the expression data, cell and gene metrics, associated metadata, and, in the case of multimodal data like ADTs from CITE-seq experiments, data from additional cell-based assays.
```
I feel like we should indicate more clearly that ADT & RNA data are in separate hdf5 files, but maybe this section isn't the right spot for that information!

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -1,21 +1,23 @@
-This object contains the expression data, cell and gene metrics, associated metadata, and, in the case of multimodal data like ADTs from CITE-seq experiments, data from additional cell-based assays.
+Single-cell or single-nuclei gene expression data (unfiltered, filtered, or processed) is provided for use with R as an RDS file containing a [`SingleCellExperiment` object](http://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html) or with python as an HDF5 file containing an [`AnnData` object](https://anndata.readthedocs.io/en/latest/index.html).
+These objects contain the expression data, cell and gene metrics, associated metadata, and, in the case of multimodal data like ADTs from CITE-seq experiments, data from additional cell-based assays.
+Note that multiplexed sample libraries are only available as `SingleCellExperiment` objects, and are not currently available as `AnnData` objects.
```
(if emojis are kosher for our style here!)

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -296,3 +298,165 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+| `development_stage_ontology_term_id` | [`HsapDv` ontology](http://obofoundry.org/ontology/hsapdv.html) term indicating developmental stage. If unavailable, `unknown` is used.  |
+| `sex_ontology_term_id` | [`PATO`](http://obofoundry.org/ontology/pato.html) term referring to the sex of the sample. If unavailable, `unknown` is used. |
+| `organism_ontology_id` | [NCBI taxonomy](https://www.ncbi.nlm.nih.gov/taxonomy) term for organism, e.g. [`NCBITaxon:9606`](http://purl.obolibrary.org/obo/NCBITaxon_9606). |
+| `self_reported_ethnicity_ontology_term_id` | For _Homo sapiens_, a [`Hancestro` term](http://obofoundry.org/ontology/hancestro.html). `multiethnic` indicates more than one ethnicity is reported. `unknown` indicates unavailable ethnicity and `NA` is used for all other organisms.  |
```
```suggestion
| `self_reported_ethnicity_ontology_term_id` | For _Homo sapiens_, a [`Hancestro` term](http://obofoundry.org/ontology/hancestro.html). `multiethnic` indicates more than one ethnicity is reported. `unknown` indicates unavailable ethnicity, and `NA` is used for all other organisms.  |
```

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -296,3 +298,165 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+| `self_reported_ethnicity_ontology_term_id` | For _Homo sapiens_, a [`Hancestro` term](http://obofoundry.org/ontology/hancestro.html). `multiethnic` indicates more than one ethnicity is reported. `unknown` indicates unavailable ethnicity and `NA` is used for all other organisms.  |
+| `disease_ontology_term_id` | [`MONDO`](http://obofoundry.org/ontology/mondo.html) term indicating disease type. [`PATO:0000461`](http://purl.obolibrary.org/obo/PATO_0000461) indicates normal or healthy tissue. If unavailable, `NA` is used.  |
+| `tissue_ontology_term_id`| [`UBERON`](http://obofoundry.org/ontology/uberon.html) term indicating tissue of origin. If unavailable, `NA` is used. |
+| `is_primary_data` | Set to `FALSE` for all libraries to reflect that all libraries were obtained from external investigators.Required by `CEllxGENE`             |
```
That said, I feel like this doesn't belong in our docs? Or if it does, a lot more columns are also required by czi!

sjspielman commented on docs/sce_file_contents.md:
````diff
@@ -296,3 +298,165 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
 
 Genetic demultiplexing statistics are found in the main `colData(sce)` data frame, with the prefix `vireo_`.
 
+## Components of an `AnnData` object
+
+Before getting started, we highly encourage you to familiarize yourself with the general `AnnData` object structure and functions available as part of the [`AnnData` package](https://anndata.readthedocs.io/en/latest/index.html).
+For the most part, the `AnnData` objects that we provide are formatted to match the expected data format for [`CELLxGENE`](https://cellxgene.cziscience.com/) following [schema version `3.0.0`](https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/3.0.0/schema.md).
+
+To begin, you will need to load the `AnnData` package and read the HDF5 file:
+
+```python
+import anndata
+adata_object = anndata.read_h5ad("SCPCL000000_processed.rds")
+```
````
don't these have an `_rna` suffix?

sjspielman commented on docs/sce_file_contents.md:
````diff
@@ -296,3 +298,165 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+
+Finally, additional metadata for ADT processing can be found in the `.uns` slot of the `AnnData` object.
+This metadata slot has the same contents as the [RNA experiment metadata](#anndata-experiment-metadata), along with one additional field, `ambient_profile`, which holds a list of representing the ambient concentrations of each ADT.
+
+```r
+adt_metadata <- metadata(altExp(sce, "adt"))
+```
````
this isn't anndata

sjspielman reviewed (commented):
Looks like you made a real dent in these docs! I've left a first round of initial comments, but I plan to come back tomorrow with fresher eyes (I'm a little screen-fatigued at the moment) and do a more thorough review.

allyhawkins commented:
Thanks for taking an initial look @sjspielman! I incorporated some of your suggestions and then I also went through all the code chunks and made sure we always assign to a variable in both SCE and AnnData experiments. 

I also added two sentences stating that CITE-seq data is included with the main RNA data in SCE but separate in AnnData. I wasn't totally sure about this being here, but I agree that I think it's important to state somewhere.

sjspielman commented:
quick FYI re https://github.com/AlexsLemonade/scpca-docs/pull/157/commits/5c04822569c95fc9198a9a089a02f1b2102ee31b

You can tell spellcheck to like emojis :) - https://github.com/AlexsLemonade/training-modules/blob/9af9931b8a99723cce9e896648fb60a66403497c/scripts/spell-check.R#L12-L13

jashapiro commented on components/dictionary.txt:
```diff
@@ -33,6 +34,7 @@ Genomics
 github
 GRCh
 hashedDrops
+HDF
```
Just curiousity, but do we ever use bare `HDF` and what happens if we put `HDF5` in the dirctionary?

jashapiro commented on docs/sce_file_contents.md:
```diff
@@ -1,21 +1,28 @@
 
+## Components of a `SingleCellExperiment` object
+
+Before getting started, we highly encourage you to familiarize yourself with the general `SingleCellExperiment` object structure and functions available as part of the [`SingleCellExperiment` package](https://bioconductor.org/packages/3.13/bioc/html/SingleCellExperiment.html) from Bioconductor.
```
I just updated to `3.17`, which is what we are using now, but maybe `release` so we don't have to update it again? We should generally make sure any other bioconductor links are consistent.
```suggestion
Before getting started, we highly encourage you to familiarize yourself with the general `SingleCellExperiment` object structure and functions available as part of the [`SingleCellExperiment` package](https://bioconductor.org/packages/3.17/bioc/html/SingleCellExperiment.html) from Bioconductor.
```

jashapiro commented on docs/sce_file_contents.md:
````diff
@@ -170,32 +177,32 @@ The list of highly variable genes used was selected using `scran::modelGeneVar`
 
 ```r
-reducedDim(sce, "PCA")
+pca <- reducedDim(sce, "PCA")
````
I don't know how I feel about the assignment additions. I feel like the bare command was sufficient and probably preferred in many contexts.

jashapiro commented on docs/sce_file_contents.md:
```diff
@@ -231,26 +238,26 @@ This data frame contains the following columns with statistics for each ADT:
 Column names are again cell barcode sequences and row names the HTO ids for all assayed HTOs.
 
-The following additional per-cell data columns for the cellhash data can be found in the main `colData` data frame (accessed with `colData(sce)` [as above](#cell-metrics)).
+The following additional per-cell data columns for the cellhash data can be found in the main `colData` data frame (accessed with `colData(sce)` [as above](#singlecellexperiment-singlecellexperiment-cell-metrics)).
```
```suggestion
The following additional per-cell data columns for the cellhash data can be found in the main `colData` data frame (accessed with `colData(sce)` [as above](#singlecellexperiment-cell-metrics)).
```

jashapiro commented on docs/sce_file_contents.md:
```diff
@@ -170,32 +177,32 @@ The list of highly variable genes used was selected using `scran::modelGeneVar`
 Column names are again cell barcode sequences and row names are the antibody targets for each ADT.
 
-Note that only cells which are denoted as "Keep" in  the `colData(sce)$adt_scpca_filter` column (as described [above](#cell-metrics)) have normalized expression values in the `logcounts` assay, and all other cells are assigned `NA` values.
+Note that only cells which are denoted as "Keep" in  the `colData(sce)$adt_scpca_filter` column (as described [above](#singlecellexperiment-singlecellexperiment-cell-metrics)) have normalized expression values in the `logcounts` assay, and all other cells are assigned `NA` values.
```
Looks like some double `singlecellexperiment-` insertions happened?
```suggestion
Note that only cells which are denoted as "Keep" in  the `colData(sce)$adt_scpca_filter` column (as described [above](#singlecellexperiment-cell-metrics)) have normalized expression values in the `logcounts` assay, and all other cells are assigned `NA` values.
```

jashapiro commented on docs/sce_file_contents.md:
```diff
@@ -170,32 +177,32 @@ The list of highly variable genes used was selected using `scran::modelGeneVar`
 However, as described in the {ref}`processed ADT data section <processing_information:Processed ADT data>`, normalization may fail under certain circumstances, in which case there will be no `logcounts` normalized expression matrix present in the alternative experiment.
 
-The following additional per-cell data columns for the ADT data can be found in the main `colData` data frame (accessed with `colData(sce)` [as above](#cell-metrics)).
+The following additional per-cell data columns for the ADT data can be found in the main `colData` data frame (accessed with `colData(sce)` [as above](#singlecellexperiment-singlecellexperiment-cell-metrics)).
```
```suggestion
The following additional per-cell data columns for the ADT data can be found in the main `colData` data frame (accessed with `colData(sce)` [as above](#singlecellexperiment-cell-metrics)).
```

jashapiro commented on docs/sce_file_contents.md:
````diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+
+```python
+import anndata
+adata_object = anndata.read_h5ad("SCPCL000000_processed_rna.hdf5")
````
This assignment makes sense, but I would leave out the others.

jashapiro commented on docs/sce_file_contents.md:
````diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+The `X` matrix can be accessed with the following python code:
+
+```python
+raw_count_matrix = adata_obj.X
````
I think I might mark the values here and below as comments and leave off the assignment.
```suggestion
adata_obj.X # raw count matrix
```

jashapiro commented on docs/sce_file_contents.md:
````diff
@@ -170,32 +177,32 @@ The list of highly variable genes used was selected using `scran::modelGeneVar`
 
 ```r
-reducedDim(sce,"UMAP")
+umap <- reducedDim(sce,"UMAP")
````
Noting missing space, but I would lose the `umap <- ` as well per earlier comments.
```suggestion
umap <- reducedDim(sce, "UMAP")
```

jashapiro commented on docs/sce_file_contents.md:
````diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+
+Metadata associated with {ref}`data processing <processing_information:Processing information>` is included in the `.uns` slot as a list.
+
+```r
````
````suggestion
```python
````

sjspielman commented:
> We probably want to update all links to bioconductor to 3.17 or release if we think we are updating more frequently.

@jashapiro FYI I had a similar thought when reviewing and opened  - https://github.com/AlexsLemonade/scpca-docs/issues/158

allyhawkins commented:
> I think I would favor line comments in both the R and python code over assignment. Much of the time people might not need to assign internal components to new variables, so I don't think we need to show it as part of the explanation.

I removed the assignments across the board and replaced them with comments. I did leave the assignment of both the sce and AnnData object when reading them in though. 

> - We probably want to update all links to bioconductor to 3.17 or release if we think we are updating more frequently.

> Finally: are we taking advantage of the ability to create versions in readthedocs? This is related to the second point above, but if docs change depending on the version of the workflow run, we might want to keep multiple versions around. Probably something to explore later.

For now, I updated this to 3.17 in this doc. Right now we don't have a process for releasing and adding version tags to the docs, and we probably want to.

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+| `development_stage_ontology_term_id` | [`HsapDv` ontology](http://obofoundry.org/ontology/hsapdv.html) term indicating developmental stage. If unavailable, `unknown` is used.  |
+| `sex_ontology_term_id` | [`PATO`](http://obofoundry.org/ontology/pato.html) term referring to the sex of the sample. If unavailable, `unknown` is used. |
+| `organism_ontology_id` | [NCBI taxonomy](https://www.ncbi.nlm.nih.gov/taxonomy) term for organism, e.g. [`NCBITaxon:9606`](http://purl.obolibrary.org/obo/NCBITaxon_9606). |
+| `self_reported_ethnicity_ontology_term_id` | For _Homo sapiens_, a [`Hancestro` term](http://obofoundry.org/ontology/hancestro.html). `multiethnic` indicates more than one ethnicity is reported. `unknown` indicates unavailable ethnicity, and `NA` is used for all other organisms.  |
```
```suggestion
| `self_reported_ethnicity_ontology_term_id` | For _Homo sapiens_, a [`HANCESTRO` term](http://obofoundry.org/ontology/hancestro.html). `multiethnic` indicates more than one ethnicity is reported. `unknown` indicates unavailable ethnicity, and `NA` is used for all other organisms.  |
```

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+| `sex_ontology_term_id` | [`PATO`](http://obofoundry.org/ontology/pato.html) term referring to the sex of the sample. If unavailable, `unknown` is used. |
+| `organism_ontology_id` | [NCBI taxonomy](https://www.ncbi.nlm.nih.gov/taxonomy) term for organism, e.g. [`NCBITaxon:9606`](http://purl.obolibrary.org/obo/NCBITaxon_9606). |
+| `self_reported_ethnicity_ontology_term_id` | For _Homo sapiens_, a [`Hancestro` term](http://obofoundry.org/ontology/hancestro.html). `multiethnic` indicates more than one ethnicity is reported. `unknown` indicates unavailable ethnicity, and `NA` is used for all other organisms.  |
+| `disease_ontology_term_id` | [`MONDO`](http://obofoundry.org/ontology/mondo.html) term indicating disease type. [`PATO:0000461`](http://purl.obolibrary.org/obo/PATO_0000461) indicates normal or healthy tissue. If unavailable, `NA` is used.  |
```
```suggestion
| `disease_ontology_term_id` | [`Mondo`](http://obofoundry.org/ontology/mondo.html) term indicating disease type. [`PATO:0000461`](http://purl.obolibrary.org/obo/PATO_0000461) indicates normal or healthy tissue. If unavailable, `NA` is used.  |
```

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+| `organism_ontology_id` | [NCBI taxonomy](https://www.ncbi.nlm.nih.gov/taxonomy) term for organism, e.g. [`NCBITaxon:9606`](http://purl.obolibrary.org/obo/NCBITaxon_9606). |
+| `self_reported_ethnicity_ontology_term_id` | For _Homo sapiens_, a [`Hancestro` term](http://obofoundry.org/ontology/hancestro.html). `multiethnic` indicates more than one ethnicity is reported. `unknown` indicates unavailable ethnicity, and `NA` is used for all other organisms.  |
+| `disease_ontology_term_id` | [`MONDO`](http://obofoundry.org/ontology/mondo.html) term indicating disease type. [`PATO:0000461`](http://purl.obolibrary.org/obo/PATO_0000461) indicates normal or healthy tissue. If unavailable, `NA` is used.  |
+| `tissue_ontology_term_id`| [`UBERON`](http://obofoundry.org/ontology/uberon.html) term indicating tissue of origin. If unavailable, `NA` is used. |
```
```suggestion
| `tissue_ontology_term_id`| [`Uberon`](http://obofoundry.org/ontology/uberon.html) term indicating tissue of origin. If unavailable, `NA` is used. |
```

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+
+| Item name   | Contents                                                         |
+| ------------- | ---------------------------------------------------------------- |
+| `schema_version` | CZI schema version used for `AnnData` formatting, `3.0.0` |
```
can't hurt?

```suggestion
| `schema_version` | CZI schema version used for `AnnData` formatting, [`3.0.0`](https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/3.0.0/schema.md) |
```

jashapiro replied:
I'd actually just eliminate the version number here, since that is the value inside.
```suggestion
| `schema_version` | CZI schema version used for `AnnData` formatting |
```

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+
+### Additional `AnnData` components for CITE-seq libraries (with ADT tags)
+
+ADT data from CITE-seq experiments, when present, is available as a separate `AnnData` object (`.hdf5` file).
```
Elsewhere we write `HDF5 file` -

```suggestion
ADT data from CITE-seq experiments, when present, is available as a separate `AnnData` object (HDF5 file).
```

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+ADT data from CITE-seq experiments, when present, is available as a separate `AnnData` object (`.hdf5` file).
+All files containing ADT data will contain the `_adt.hdf5` suffix.
+
+The data matrix, `X`, of the `AnnData` objects contain the primary ADT expression data as integer counts in both the unfiltered (`_unfiltered_rna.hdf5`) and filtered (`_filtered_rna.hdf5`) objects.
```
```suggestion
The data matrix, `X`, of the `AnnData` objects contain the primary ADT expression data as integer counts in both the unfiltered (`_unfiltered_adt.hdf5`) and filtered (`_filtered_adt.hdf5`) objects.
```

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+
+### `AnnData` dimensionality reduction results
+
+In the HDF5 file containing the processed `SingleCellExperiment` object only (`_processed_rna.hdf5`), the `.obsm` slot of the object will be occupied with both principal component analysis (`X_PCA`) and `X_UMAP` results.
```
```suggestion
The HDF5 file containing the processed `AnnData` object (`_processed_rna.hdf5`) contains a slot `.obsm` with both principal component analysis (`X_PCA`) and UMAP (`X_UMAP`) results.
```

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+All files containing ADT data will contain the `_adt.hdf5` suffix.
+
+The data matrix, `X`, of the `AnnData` objects contain the primary ADT expression data as integer counts in both the unfiltered (`_unfiltered_rna.hdf5`) and filtered (`_filtered_rna.hdf5`) objects.
+Each column corresponds to a cell or droplet (in the same order as the main `AnnData` object) and each row corresponds to an antibody derived tag (ADT).
```
```suggestion
Each column corresponds to a cell or droplet (in the same order as the main `AnnData` object), and each row corresponds to an antibody derived tag (ADT).
```

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+Column names are again cell barcode sequences and row names are the antibody targets for each ADT.
+
+As with the RNA `AnnData` objects, in processed objects _only_ (`_processed_adt.hdf5`), the data matrix `X` contains the normalized ADT counts and the primary data can be found in `raw.X`.
+Note that only cells which are denoted as "Keep" in  the `adata_obj.uns["adt_scpca_filter"]` column (as described [above](#singlecellexperiment-cell-metrics)) have normalized expression values in the `logcounts` assay, and all other cells are assigned `NA` values.
```
```suggestion
Note that only cells which are denoted as `"Keep"` in the `adata_obj.uns["adt_scpca_filter"]` column (as described [above](#singlecellexperiment-cell-metrics)) have normalized expression values in the `logcounts` assay, and all other cells are assigned `NA` values.
```

sjspielman replied:
**BUT** - we're in anndata land so there is no `logcounts` assay, so this text needs to change beyond my suggestion

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+
+As with the RNA `AnnData` objects, in processed objects _only_ (`_processed_adt.hdf5`), the data matrix `X` contains the normalized ADT counts and the primary data can be found in `raw.X`.
+Note that only cells which are denoted as "Keep" in  the `adata_obj.uns["adt_scpca_filter"]` column (as described [above](#singlecellexperiment-cell-metrics)) have normalized expression values in the `logcounts` assay, and all other cells are assigned `NA` values.
+However, as described in the {ref}`processed ADT data section <processing_information:Processed ADT data>`, normalization may fail under certain circumstances, in which case there will be no normalized expression matrix present in the `AnnData` object and the primary data will still be stored in `X`.
```
```suggestion
However, as described in the {ref}`processed ADT data section <processing_information:Processed ADT data>`, normalization may fail under certain circumstances.
In such cases the `AnnData` object will not contain a normalized expression matrix, but the primary data will still be stored in `X`.
```

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+Note that only cells which are denoted as "Keep" in  the `adata_obj.uns["adt_scpca_filter"]` column (as described [above](#singlecellexperiment-cell-metrics)) have normalized expression values in the `logcounts` assay, and all other cells are assigned `NA` values.
+However, as described in the {ref}`processed ADT data section <processing_information:Processed ADT data>`, normalization may fail under certain circumstances, in which case there will be no normalized expression matrix present in the `AnnData` object and the primary data will still be stored in `X`.
+
+In addition, the following QC statistics from [`DropletUtils::cleanTagCounts()`](https://rdrr.io/github/MarioniLab/DropletUtils/man/cleanTagCounts.html) can be found in the `obs` of the `AnnData` object.
```
```suggestion
In addition, the following QC statistics from [`DropletUtils::cleanTagCounts()`](https://rdrr.io/github/MarioniLab/DropletUtils/man/cleanTagCounts.html) can be found in the `obs` slot of each ADT-specific `AnnData` object.
```

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+
+Metrics for each of the ADTs assayed can be found as a `pandas.DataFrame` in the `.var` slot within the `AnnData` object:
+
+This data frame contains the following columns with statistics for each ADT:
```
i think you can delete this as duplicative with previous sentence?

```suggestion
```

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -296,3 +303,161 @@ For methods that rely on the HTO data, these statistics are found in the `colDat
+| `target_type` | Whether each ADT is a target (`target`), negative/isotype control (`neg_control`), or positive control (`pos_control`). If this information was not provided, all ADTs will have been considered targets and will be labeled as `target` |
+
+Finally, additional metadata for ADT processing can be found in the `.uns` slot of the `AnnData` object.
+This metadata slot has the same contents as the [RNA experiment metadata](#anndata-experiment-metadata), along with one additional field, `ambient_profile`, which holds a list of representing the ambient concentrations of each ADT.
```
```suggestion
This metadata slot has the same contents as the [RNA experiment metadata](#anndata-experiment-metadata), along with one additional field, `ambient_profile`, which holds a list of the ambient concentrations of each ADT.
```

sjspielman commented on docs/sce_file_contents.md:
```diff
@@ -231,26 +238,26 @@ This data frame contains the following columns with statistics for each ADT:
 
 Finally, additional metadata for ADT processing can be found in the metadata slot of the alternative experiment.
-This metadata slot has the same contents as the [parent experiment metadata](#experiment-metadata), along with one additional field, `ambient_profile`, which holds a list of representing the ambient concentrations of each ADT.
+This metadata slot has the same contents as the [parent experiment metadata](#singlecellexperiment-experiment-metadata), along with one additional field, `ambient_profile`, which holds a list of representing the ambient concentrations of each ADT.
```
```suggestion
This metadata slot has the same contents as the [parent experiment metadata](#singlecellexperiment-experiment-metadata), along with one additional field, `ambient_profile`, which holds a list of the ambient concentrations of each ADT.
```

sjspielman reviewed (commented):
Left some more careful textual comments!

allyhawkins commented:
I applied the suggestions and addressed the small comments from both of you. This is ready for another look @jashapiro.

jashapiro reviewed (approved):
LGTM
